### PR TITLE
[WIP] Handle home_path setting

### PR DESCRIPTION
**Pull request by broskoski**
This basically works, just needs to handle the custom home_path stuff. 

Makes me think though that maybe `/feed` should be the default setting on the attribute in the API?

### CODE REVIEW

**Comment by broskoski** [comment]
@dzucconi what say you

File: apps/feed/index.coffee [hunk 1]
```diff
@@ -6,10 +6,18 @@ app = module.exports = express()
 app.set "views", __dirname + "/templates"
 app.set "view engine", "jade"
 
-app.get "/", routes.index
+homePathMiddleware = (req, res, next) -> 
+  return next() unless req.user
+  
+  path = req.user.get('home_path') 
+  return res.redirect 302, path unless path is "/" or !path
```
Comment (dzucconi) [inline apps/feed/index.coffee]: `!path` will never be evaluated, if it's `null` it's not going to be `'/'`

File: apps/settings/templates/tabs/general.jade [hunk 2]
```diff
@@ -61,6 +61,26 @@ block content
...
+
+                option(
+                  value='/explore'
+                  selected= user.get('home_path') == '/explore'
+                ) Explore
+
+                option(
+                  value="/#{user.get('slug')}"
+                  selected= user.get('home_path') == "/#{user.get('slug')}"
+                ) Profile
```
Comment (dzucconi) [inline apps/settings/templates/tabs/general.jade]: How do you set a custom one?

File: lib/middleware/locals.coffee [hunk 3]
```diff
@@ -10,6 +10,7 @@ _ = require 'underscore'
 module.exports = (req, res, next) ->
   res.locals.sd.CURRENT_PATH = parse(req.url).pathname
   res.locals._ = _
-  res.locals.homeHref = if parseInt(req.user?.get('following_count')) <= 1 then '/explore' else '/'
+  console.log('req.user?.homePath()', req.user?.homePath())
```
Comment (dzucconi) [inline lib/middleware/locals.coffee]: log

File: apps/settings/templates/tabs/general.jade (hunk 2 above)
Comment (broskoski) [inline apps/settings/templates/tabs/general.jade]: Still need to handle that, hence the WIP

File: models/user.coffee [hunk 4]
```diff
@@ -22,6 +22,9 @@ module.exports = class User extends Model
   isPremium: ->
     @get('is_premium')
 
+  homePath: ->
+    if parseInt(@get('following_count')) <= 1 then '/explore' else (@get('home_path') or '/')
```
Comment (dzucconi) [inline models/user.coffee]: OMG I never put it together that this was the behavior and was always confused.

**Comment by dzucconi** [comment]
I think this approach with the middleware is 👌

File: apps/feed/index.coffee (hunk 1 above)
Comment (dzucconi) [inline apps/feed/index.coffee]: Oh, I misread this. Still think it's a confusing condition.

Comment (broskoski) [inline apps/feed/index.coffee]: Yeah agreed, was trying to be extra careful to have it still evaluate even if the user doesn't have a `home_path` set. I think too careful to the point of being confusing.

Comment (dzucconi) [inline apps/feed/index.coffee]:
```coffeescript
return next() unless req.user.has('home_path')
```

instead of collapsing those different conditions

Comment (dzucconi) [inline apps/feed/index.coffee]: Does a req always have a user model, even if you're logged out?

Comment (broskoski) [inline apps/feed/index.coffee]: Good call. No, if you're logged out no user model will be present.

**Pull request closed without merging**